elasticsearch/client.go

```diff
@@ -0,0 +1,133 @@
+package elasticsearch
+
+import (
+	"bytes"
+	"context"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+
+	dphttp "github.com/ONSdigital/dp-net/http"
+	"github.com/ONSdigital/log.go/log"
+	awsauth "github.com/smartystreets/go-aws-auth"
+)
+
+// ServiceName elasticsearch
+const ServiceName = "elasticsearch"
+
+// Client is an ElasticSearch client containing an HTTP client to contact the elasticsearch API.
+type Client struct {
+	httpCli      dphttp.Clienter
+	url          string
+	serviceName  string
+	signRequests bool
+	indexes      []string
+}
+
+// NewClient returns a new initialised elasticsearch client with the default dp-net/http client
+func NewClient(url string, signRequests bool, maxRetries int, indexes ...string) *Client {
+	httpClient := dphttp.NewClient()
+	httpClient.SetMaxRetries(maxRetries)
+	return NewClientWithHTTPClient(url, signRequests, httpClient, indexes...)
+}
+
+// NewClientWithHTTPClient returns a new initialised elasticsearch client with the provided HTTP client
+func NewClientWithHTTPClient(url string, signRequests bool, httpCli dphttp.Clienter, indexes ...string) *Client {
+	return &Client{
+		httpCli:      httpCli,
+		url:          url,
+		serviceName:  ServiceName,
+		signRequests: signRequests,
+		indexes:      indexes,
+	}
+}
+
+//CreateIndex creates an index in elasticsearch
+func (cli *Client) CreateIndex(ctx context.Context, indexName string, indexSettings ...byte) (int, error) {
```

Comment:
I would keep `indexSettings` as type `[]byte` - as this is more consistent with the other functions, and it also allows a nil value to be passed if needed.

Reply:
Thank you, that is a great idea